src/main/kotlin/com/openlattice/mechanic/upgrades/CreateAtlasUsersAndSetPermissions.kt

```diff
@@ -0,0 +1,262 @@
+package com.openlattice.mechanic.upgrades
+
+import com.hazelcast.query.Predicates
+import com.openlattice.assembler.AssemblerConfiguration
+import com.openlattice.assembler.AssemblerConnectionManager
+import com.openlattice.assembler.MEMBER_ORG_DATABASE_PERMISSIONS
+import com.openlattice.assembler.PostgresDatabases
+import com.openlattice.assembler.PostgresRoles.Companion.buildPostgresUsername
+import com.openlattice.authorization.*
+import com.openlattice.authorization.mapstores.PermissionMapstore
+import com.openlattice.authorization.securable.SecurableObjectType
+import com.openlattice.directory.MaterializedViewAccount
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.organization.OrganizationExternalDatabaseColumn
+import com.openlattice.organization.OrganizationExternalDatabaseTable
+import com.openlattice.organizations.Organization
+import com.openlattice.postgres.DataTables
+import com.openlattice.postgres.PostgresPrivileges
+import com.zaxxer.hikari.HikariDataSource
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class CreateAtlasUsersAndSetPermissions(
+        private val toolbox: Toolbox,
+        private val assemblerConfiguration: AssemblerConfiguration
+) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(CreateAtlasUsersAndSetPermissions::class.java)
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2020_09_15.value
+    }
+
+    override fun upgrade(): Boolean {
+        val dbCreds = HazelcastMap.DB_CREDS.getMap(toolbox.hazelcast).toMap()
+        val orgs = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast).toMap()
+
+        val principalsToAccounts = HazelcastMap.PRINCIPALS.getMap(toolbox.hazelcast)
+                .values
+                .toSet()
+                .associate {
+                    val dbUserId = buildPostgresUsername(it)
+                    it.principal to dbCreds.getValue(dbUserId)
+                }
+
+        configureUsersInAtlas(principalsToAccounts.filter { it.key.type == PrincipalType.USER }.values)
```

Comment:
This should rename user accounts if they already exist before creating the user. Given this we should merge this with previous migration to assign a new user id, starting the counter at 0, and  Also that is we should reset the user counter to 0 and assign user ids as part of this migration.